py4cast/lightning.py

```diff
@@ -36,6 +42,58 @@
 PLOT_PERIOD: int = 10
 
 
+@dataclass
+class PlDataModule(pl.LightningDataModule):
+    """
+    DataModule to encapsulate data splits and data loading.
+    """
+
+    dataset: str
+    num_input_steps: int
+    num_pred_steps_train: int
+    num_pred_steps_val_test: int
+    dl_settings: TorchDataloaderSettings
+    dataset_conf: Union[Path, None] = None
+    config_override: Union[Dict, None] = (None,)
+
+    def __post_init__(self):
+        super().__init__()
+
+        # Get dataset in initialisation to have access to this attribute before method trainer.fit
+        self.train_ds, self.val_ds, self.test_ds = get_datasets(
+            self.dataset,
+            self.num_input_steps,
+            self.num_pred_steps_train,
+            self.num_pred_steps_val_test,
+            self.dataset_conf,
+            self.config_override,
+        )
+
+    @property
+    def get_len_train_dl(self):
+        return len(self.train_ds.torch_dataloader(self.dl_settings))
+
+    @property
+    def get_train_dataset_info(self):
+        return self.train_ds.dataset_info
+
+    @property
+    def get_infer_ds(self):
+        return self.test_ds
```

Comment:
You should remove the **get** prefix, it is not necessary for properties and very "java like"